### PR TITLE
Add several API hooks to emulate through couple of Rust samples.

### PR DESCRIPTION
**Pull request by dmsft**
CreateSemaphoreW()
SetThreadStackGuarantee()
SetThreadDescription()
InitOnceBeginInitialize()
_errno()
fputc()
signal()

### CODE REVIEW

File: speakeasy/winenv/api/usermode/kernel32.py [hunk 1]
```diff
@@ -6064,3 +6064,49 @@ def WaitForSingleObjectEx(self, emu, argv, ctx={}):
...
+    def InitOnceBeginInitialize(self, emu, argv, ctx={}):
+        '''
+        BOOL InitOnceBeginInitialize(
+            [in, out]       LPINIT_ONCE lpInitOnce,
+            [in]            DWORD       dwFlags,
+            [out]           PBOOL       fPending,
+            [out, optional] LPVOID      *lpContext
+        );
+        '''
+        # _, dwFlags, _, _ = argv
```
Comment (williballenthin) [inline speakeasy/winenv/api/usermode/kernel32.py:6109]:
```suggestion
```

**Review by williballenthin — APPROVED** [review]

**Comment by williballenthin** [comment]
thanks!

Once the CLA is signed I'd be happy to merge these updates.

**Comment by dmsft** [comment]
CLA signed, thank you.

**Review by williballenthin — APPROVED** [review]
thank you!

**Pull request merged**